src/main/java/io/fusionauth/domain/event/UserIdentityProviderLinkEvent.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2019-2022, FusionAuth, All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+package io.fusionauth.domain.event;
+
+import com.inversoft.json.JacksonConstructor;
+import com.inversoft.json.ToString;
+import io.fusionauth.domain.Buildable;
+import io.fusionauth.domain.EventInfo;
+import io.fusionauth.domain.User;
+
+/**
+ * Models the User IdP Link Event.
+ *
+ * @author Rob Davis
+ */
+public class UserIdentityProviderLinkEvent extends BaseEvent implements Buildable<UserIdentityProviderLinkEvent>, NonTransactionalEvent {
+  public String identityProviderName;
+
+  public User user;
+
+  @JacksonConstructor
+  public UserIdentityProviderLinkEvent() {
+  }
+
+  public UserIdentityProviderLinkEvent(EventInfo info, String identityProviderName, User user) {
```

Comment:
It is fine if we add the name, but the event needs to for sure contain the IdP Id.  It seems that we should also include details about the link itself. Should we be adding the entire link object? Or at least the IdP user unique Id, IdP user display name if we have it, the FusionAuth IdP link Id, etc. We need to assume someone wants to consume this event and then use it to create FKs, or call back to FusionAuth and be able to retrieve specific information.